git --diff  a/tests/test_msgs.py b/tests/test_msgs.py
--- a/tests/test_msgs.py	2023-03-24 07:34:22.000000000 -0500
+++ b/tests/test_msgs.py	2023-10-04 15:21:40.898295624 -0500
@@ -5,6 +5,7 @@
 import time
 import unittest
 from typing import Optional
+import pydantic
 
 from commlib.msg import Message, MessageHeader, PubSubMessage, RPCMessage
 from commlib.timer import Timer
@@ -46,7 +47,10 @@
 
         _msg = TestPubSubMessage()
         _msg.b = TestObject(c=2, d=3)
-        self.assertEqual(_msg.dict(), _d)
+        if pydantic.__version__ < '2.0.0':
+            self.assertEqual(_msg.dict(), _d)
+        else:
+            self.assertEqual(_msg.model_dump(), _d)
 
     def test_nested_message_from_dict(self):
         _d = {
